core/src/main/java/io/onetable/iceberg/IcebergSchemaExtractor.java

```diff
@@ -67,7 +67,11 @@ public Schema toIceberg(OneSchema oneSchema) {
     AtomicInteger fieldIdTracker = new AtomicInteger(0);
     List<Types.NestedField> nestedFields = convertFields(oneSchema, fieldIdTracker);
     List<OneField> recordKeyFields = oneSchema.getRecordKeyFields();
-    if (recordKeyFields.isEmpty()) {
+    boolean recordKeyFieldsAreNotRequired = recordKeyFields.stream().anyMatch(f -> f.getSchema().isNullable());
```

Comment:
Add a comment explaining about Iceberg expectations and which is why this change is made.